File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
        => number == number.DigitsRec(new List<int> {}).Sum(n => Math.Pow(n, number.ToString().Length));

    public static IEnumerable<int> DigitsRec(this int n, List<int> aggregator)
    {
        if (n < 1) {
            return aggregator;
        }
        aggregator.Add(n % 10);
        return (n / 10).DigitsRec(aggregator);
    }

    public static IEnumerable<int> Digits(this int n)
    {
        while (n > 0)
        {
            yield return n % 10;
            n /= 10;
        }
    }
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.util.ArrayList;
import java.util.List;
import java.util.function.IntUnaryOperator;

class ArmstrongNumbers {

  private List<Integer> digits(List<Integer> acc, int number) {
    if (number == 0) {
      return acc;
    }
    acc.add(number % 10);
    return digits(acc, number / 10);
  }

  private int armstrong(int number) {
    List<Integer> digit = digits(new ArrayList<Integer>(), number);
    IntUnaryOperator power = n -> (int) Math.pow(n, digit.size());
    return digit.stream().mapToInt(Integer::intValue).map(power).sum();
  }

  boolean isArmstrongNumber(int numberToCheck) {
    return numberToCheck == armstrong(numberToCheck);
  }
}
